tests/aws/services/cloudwatch/test_cloudwatch_metrics.py

```diff
@@ -0,0 +1,128 @@
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING
+
+import pytest
+
+from localstack.testing.pytest import markers
+from localstack.testing.pytest.snapshot import is_aws
+from localstack.utils.strings import short_uid
+
+if TYPE_CHECKING:
+    from mypy_boto3_cloudwatch import CloudWatchClient
+from localstack.utils.sync import retry
+
+TEST_SUCCESSFUL_LAMBDA = """
+def handler(event, context):
+    return {"success": "ok"}
+"""
+
+TEST_FAILING_LAMBDA = """
+def handler(event, context):
+    raise Exception('fail on purpose')
+"""
+
+
+class TestCloudWatchLambdaMetrics:
+    """
+    Tests for metrics that are reported automatically by Lambda
+    see also https://docs.aws.amazon.com/lambda/latest/dg/monitoring-metrics.html
+    """
+
+    @markers.aws.validated
+    def test_lambda_invoke_successful(self, aws_client, create_lambda_function, snapshot):
+        """
+        successful invocation of lambda should report "Invocations" metric
+        """
+        fn_name = f"fn-cw-{short_uid()}"
+        create_lambda_function(
+            func_name=fn_name,
+            handler_file=TEST_SUCCESSFUL_LAMBDA,
+            runtime="python3.9",
+        )
+        result = aws_client.lambda_.invoke(FunctionName=fn_name)
+        assert result["StatusCode"] == 200
+        snapshot.match("invoke", result)
+
+        # wait for metrics
+        result = retry(
+            lambda: self._wait_for_lambda_metric(
+                aws_client.cloudwatch,
+                fn_name=fn_name,
+                metric_name="Invocations",
+                expected_return=[1.0],
+            ),
+            retries=200 if is_aws() else 20,
+            sleep=10 if is_aws() else 1,
+        )
+        snapshot.match("get-metric-data", result)
+
+    @pytest.mark.skipif(not is_aws(), reason="'Errors' metrics not reported by LS")
+    @markers.aws.validated
+    def test_lambda_invoke_error(self, aws_client, create_lambda_function, snapshot):
+        """
+        Unsuccessful Invocation -> resulting in error, should report
+        "Errors" and "Invocations" metrics
+        """
+        fn_name = f"fn-cw-{short_uid()}"
+        create_lambda_function(
+            func_name=fn_name,
+            handler_file=TEST_FAILING_LAMBDA,
+            runtime="python3.9",
+        )
+        result = aws_client.lambda_.invoke(FunctionName=fn_name)
+        snapshot.match("invoke", result)
+
+        # wait for metrics
+        invocation_res = retry(
+            lambda: self._wait_for_lambda_metric(
+                aws_client.cloudwatch,
+                fn_name=fn_name,
+                metric_name="Invocations",
+                expected_return=[1.0],
+            ),
+            retries=200 if is_aws() else 20,
+            sleep=10 if is_aws() else 1,
```

Comment:
and oof, that's... a long wait 😭

Reply:
you never know how long it takes on AWS 😅 but in general it's long for the metrics to show up. I didn't want to spam the AWS endpoint, so I just went for a longer period. 
you think it could become an issue? maybe once we really automated the snapshot updates 😄 ⏳

Reply:
oh no sorry, it was a general comment, I know for firehose sometimes the wait are really long, I guess that's where we shine? 😄
I guess those kind of tests is why it's hard to automate, but we can always introduce some kind of markers at that time 😄